src-lts/scripts/scr_catspeak_alloc/scr_catspeak_alloc.gml

```diff
@@ -17,6 +17,8 @@ function catspeak_collect() {
         }
         weakRef.adapter.destroy(weakRef.ds);
         array_delete(pool, i, 1);
+		--i;
```

Comment:
Can you remove these, instead iterate backwards like `for (var i = poolSize - 1; i >= 0; i -= 1)`.

Reply:
I can do that yeah, but we'll want to also decrement `poolSize` by `-1` as well. Otherwise it'll throw an error trying to access a position at size when it's size-1

Reply:
I was giving it a quick test in my own project to make sure that everything works.

Reply:
> we'll want to also subtract `poolSize` by `1` as well (only on setting up `poolSize`). Otherwise it'll throw an error trying to access a position at size when it's size-1

Hence `for (var i = poolSize - 1;` ✍(◔◡◔)

Anyway, thanks for doing that, I'll merge it!

Reply:
I won't lie, I completely forgotten how for loops have worked... I've been spending too much time in `repeat` land 😿